src/Middleware/Rewrite/src/Internal/RedirectToWwwRule.cs

```diff
@@ -32,6 +44,12 @@ public virtual void ApplyRule(RewriteContext context)
                 context.Result = RuleResult.ContinueRules;
                 return;
             }
+            
+            if (_domains.Length > 0 && !_domains.Any(d => req.Host.Host.Equals(d, StringComparison.OrdinalIgnoreCase)))
```

Comment:
Few performance ideas:
- I'd throw in the constructor if domains.Length == 0. That way, you will always have at least one domain.
- Using LINQ here seems inefficient, it's probably more performant to just have a foreach loop through each domain.

Reply:
Thanks for the review. I just pushed a commit adding the throw and replacing linq with a foreach. For further improvements we could replace:
```foreach (var domain in _domains)
{
    if (domain.Equals(req.Host.Host, StringComparison.OrdinalIgnoreCase))
    {
        domainRequiresRedirection = true;
        break;
    }
}
```
with:
```for (var i = 0; i < _domains.Length; i++)
{
    if (_domains[i].ReferenceEquals(req.Host.Host) || _domains[i].Equals(req.Host.Host, StringComparison.OrdinalIgnoreCase))
    {
        domainRequiresRedirection = true;
        break;
    }
}
```
Let me know what you'd like to see that change commited.

Reply:
I don't think _domains[i].ReferenceEquals(req.Host.Host) would ever be true, as the domain you are passing into the rule wouldn't be the same string passed into the WwwRule.

Reply:
Oh you're right! The strings will on most cases come from different instances.